### ISSUE TITLE
salt-key cmd return error

### ISSUE THREAD

**Issue by holyzhou** [saltstack/salt#19216]
in salt-ssh 2014.7.0-2017-g59622f5 (Helium) , use command   salt-key ,  i got error 

```
"Exception TypeError: 'an integer is required' in <bound method MasterEvent.__del__ of <salt.utils.event.MasterEvent object at 0x7fbb2a88d710>> ignored"
```

Happy Christmas, ^_^

**Comment by cro** [comment]
This seems to have been introduced by c2c9e56b821fa24caff6960d3e56a6becc2d5ed6.  @cachedout performed a significant refactor of the CLI components of Salt.  It appears to be harmless for the time being, we'll get it sorted next week.

**Comment by jfindlay** [comment]
This happens on the 2014.7 branch.

```
jmoney-centos-7 salt develop # salt --versions
           Salt: 2015.2.0-543-g5b7660b
         Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
 msgpack-python: 0.4.2
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.3.1
           RAET: Not Installed
            ZMQ: 3.2.4
           Mako: Not Installed
```

```
jmoney-centos-7 salt develop # salt-key
Accepted Keys:
Denied Keys:
Unaccepted Keys:
jmoney-centos-7
Rejected Keys:
Exception TypeError: 'an integer is required' in <bound method MasterEvent.__del__ of <salt.utils.event.MasterEvent object at 0x17a4b90>> ignored
```

**Comment by cachedout** [comment]
We have this fixed in develop. We are just waiting on a backport into 2015.2. I cannot reproduce this on 2014.7

**Comment by holyzhou** [comment]
@cachedout  yes , it's work fine, i test in yesterday

**Comment by rallytime** [comment]
Thanks for the confirmation @holyzhou! I'll close this out then. If this pops up again for anyone, leave a comment and we can re-visit the issue.

**Comment by gczuczy** [comment]
I also get this by using the localclient.cmd(), on 2014.7. Is this fix supposed to fix that as well?

**Comment by rallytime** [comment]
@holyzhou I think that might be a different issue. Would you mind filing a new bug?

**Comment by rodrigc** [comment]
In salt-key 2015.2.0-68-g62250b6 , if I do salt-key -L, I also see:

Exception TypeError: 'an integer is required' in <bound method MasterEvent.__del__ of <salt.utils.event.MasterEvent object at 0x80ad99290>> ignored

What was the revision in develop that fixed this?

**Comment by jordo1ken** [comment]
here you can see I marked out the hostnames to protect the innocent, but I just updated to 2015.2 with the bootstrap script from 2014.7 and now I'm getting this. The error will occur on every execution of ```salt-run``` for me. I did notice some dangling old egg packages lying around. But I think I have those cleaned up now. 

```bash
root@xxx ~ # salt-run manage.versions
Master:
    2015.2.0
Up to date:
    ----------
    xxx:
        2015.2.0
    xxx:
        2015.2.0
    xxx:
        2015.2.0
    xxx:
        2015.2.0
    xxx:
        2015.2.0
    xxx:
        2015.2.0
    xxx:
        2015.2.0
    xxx:
        2015.2.0
    xxx:
        2015.2.0
    xxx:
        2015.2.0
    xxx:
        2015.2.0
    xxx:
        2015.2.0
Exception TypeError: 'an integer is required' in <bound method MasterEvent.__del__ of <salt.utils.event.MasterEvent object at 0x2a3a990>> ignored
```

**Comment by rallytime** [comment]
@rodrigc You're right. This is still happening on 2015.2. I need to go and find the commit that fixed this so we can backport it.

@jordo1ken Are you in a place where you can test `salt-run` on the develop branch? I am curious to see if this fix that needs to be backported will also fix this issue or if something different needs to be done here.

**Comment by rallytime** [comment]
@rodrigc Ok, this has definitely been fixed on the 2015.2 branch now with #20086. And as such, I am going to close this again.

@jordo1ken I have a very high suspicion that this same pull request will clear up your problems as well. Can you give the HEAD of 2015.2 a try? If you are still seeing this problem, please feel free to file a new issue, as I believe they are separate bugs.

**Comment by jordo1ken** [comment]
@rallytime Thanks. I'll check it out.

**Comment by rodrigc** [comment]
I updated to: salt (2015.2.0rc1-85-g8e81684)
I ran salt-key -L, and it worked fine.  The problem is gone.
Thanks!!

**Comment by rallytime** [comment]
Great! Thanks!

**Comment by csakoda** [comment]
Unrelated to the main cause in this bug I think, but I got the same error when I mistakenly included circular references in some custom runners I wrote.
